build_tools/github_actions/cancel_workflow_and_wait.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Doing best-effort to cancel a running workflow (started before this script is
+# run) and wait for the cancellation to complete.
+#
+# Due to the limitations of Github API, there are race conditions cases that
+# this script can't be sure if the run is cancelled. It will detect and fail
+# fast in such cases.
```

Comment:
nits:
* Empty line between comment blocks (copyright/license header, script description)
* Rephrase description (grammar)

```suggestion
# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

# Perform a best-effort attempt to cancel a running workflow (started before
# this script is run) and wait for the cancellation to complete.
#
# Due to limitations in GitHub's APIs, there are race conditions cases that
# this script can't be sure if the run is cancelled. It will detect and fail
# fast in such cases.
```